src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -523,14 +530,15 @@ def create_service_principal_for_rbac(name=None, password=None, years=1, #pylint
             'appId': app_id,
             'password': password,
             'name': name,
+            'displayName': app_display_name,
             'tenant': graph_client.config.tenant_id
             }
     return result
 
 def reset_service_principal_credential(name, password=None, years=1):
     '''reset credential, on expiration or you forget it.
 
-    :param str name: the uri representing the name of the service principal
+    :param str name: the name, can be the app id uri, app id guid, or display name
```

Comment:
If this can be a name/id/guid, is "name" really the right name for this parameter?  Maybe "identifier" or something.